backends-clickhouse/src/main/scala/io/glutenproject/backendsapi/clickhouse/CHSparkPlanExecApi.scala

```diff
@@ -750,4 +750,8 @@ class CHSparkPlanExecApi extends SparkPlanExecApi {
   ): GenerateExecTransformerBase = {
     CHGenerateExecTransformer(generator, requiredChildOutput, outer, generatorOutput, child)
   }
+
+  override def genPreProjectForGenerate(generate: GenerateExec): SparkPlan = generate
```

Comment:
The logic of pull out should also apply to CH? It's just that CH hasn't encountered a scenario that requires pull out yet?

Reply:
So far, CH doesn't need pre/post project for their already supported generator type. But not sure if it would be needed in the future with more `Generator` supported.

Reply:
`Explode` and `PosExplode` are already supported in CH, and pre/post projection are not needed. Looks like we cannot unify the implementation.

Reply:
At least for pre-project, CH and Velox are consistent; they can only execute expressions within the project operator. I think that the logic of pre-project can also be applied to CH at present, it's just that CH has not yet started to adapt `GeneratorExec` that include expressions. I think you can put `GeneratorExec` into `needsPreProject` for validation, and only apply the pull-out logic when the child is not an `Attribute`. This should not affect the current CH, and later, if CH wants to adapt GeneratorExec` with expressions, it can be directly reused.

Reply:
`GenerateExec` was initially implemented for CH, and they should have the native implementation for the generator expressions. Please check https://github.com/apache/incubator-gluten/pull/574

Velox backend does not have the native function implementation to the generator expressions. But it has an `UnnestNode` operator that can help to generate the expected output. That's why a pre/post projection is added to adopt the input/output for the `UnnestNode`.

I believe CH doesn't require the pre/post project logic for the supported generator. cc @taiyang-li

Reply:
ch doesn't need it for now. thanks for reminding me